addons/web/static/src/views/list/list_renderer.scss

```
@@ -271,11 +271,27 @@
             cursor: col-resize;
         }
 
-        .o_list_actions_header {
+        .o_list_actions_header, .o_optional_columns_dropdown {
             width: 32px;
             background-color: var(--ListRenderer-thead-bg-color);
         }
 
+        .o_optional_columns_dropdown {
+            z-index: 2;
+            padding: var(--ListRenderer-thead-padding-v) var(--ListRenderer-table-padding-h) calc(var(--ListRenderer-thead-padding-v) + #{$form-check-margin-bottom}) $table-cell-padding-x-sm;
+
+            // Reset the styling applied by table > x
+            .dropdown-item, .dropdown-divider {
+                padding: 0;
+                border-bottom-width: 0;
+                box-shadow: none;
+            }
```

I can see why you did it :) (https://github.com/odoo/odoo/blob/saas-16.1/addons/web/static/lib/bootstrap/scss/_tables.scss#L26)

But `box-shadow` uses `--table-accent-bg` which is `transparent` and there is no border-style or border-color, so the border is not rendered.

Except keeping the padding reset for the divider element, the rest shouldn't affect the design of dropdown items & dividers.

I would do something like this :
```suggestion
            // Reset the necessary styling applied by table > x
           .dropdown-divider {
                padding: 0;
            }
```